File: packages/python/src/ai37_agent_sdk/testing/fixtures.py
from __future__ import annotations

from dataclasses import replace
from typing import Any

from ..billing.types import (
    BillingRuntimeFeature,
    BillingRuntimePrivilege,
    BillingRuntimeState,
)

_BASE = BillingRuntimeState(
    org_id="org-test",
    billing_org_id="billing-org-test",
    entitlement_status="active",
    remaining_total_tokens=1000,
    features=[],
    llm_key="sk-test-llm",
    stale=False,
)


def _active(**overrides: Any) -> BillingRuntimeState:
    return replace(_BASE, **overrides)


def _no_resources(**overrides: Any) -> BillingRuntimeState:
    return replace(
        _BASE, entitlement_status="no_resources", remaining_total_tokens=0, **overrides
    )


def _trial(**overrides: Any) -> BillingRuntimeState:
    return replace(
        _BASE,
        current_subscription_status="trialing",
        trial_ends_at="2030-01-01T00:00:00Z",
        **overrides,
    )


def _feature_grant(feature: str, privilege: str, value: bool) -> BillingRuntimeFeature:
    return BillingRuntimeFeature(
        code=feature,
        privileges=[
            BillingRuntimePrivilege(code=privilege, value=value, value_type="boolean", config={})
        ],
    )


def _feature_allowed(feature: str, privilege: str, **overrides: Any) -> BillingRuntimeState:
    return replace(_BASE, features=[_feature_grant(feature, privilege, True)], **overrides)


def _feature_denied(feature: str, privilege: str, **overrides: Any) -> BillingRuntimeState:
    return replace(_BASE, features=[_feature_grant(feature, privilege, False)], **overrides)


class _RuntimeStateFixtures:
    active = staticmethod(_active)
    no_resources = staticmethod(_no_resources)
    trial = staticmethod(_trial)
    feature_allowed = staticmethod(_feature_allowed)
    feature_denied = staticmethod(_feature_denied)


class _Fixtures:
    runtime_state = _RuntimeStateFixtures()


fixtures = _Fixtures()

File: packages/ts/src/testing/fixtures.ts
// Канонические фикстуры BillingRuntimeState для тестов агентов (по contract/).
import type {
  BillingRuntimeFeature,
  BillingRuntimeState,
} from '../billing/types'

const BASE: BillingRuntimeState = {
  orgId: 'org-test',
  billingOrgId: 'billing-org-test',
  entitlementStatus: 'active',
  remainingTotalTokens: 1000,
  features: [],
  llmKey: 'sk-test-llm',
  stale: false,
}

function active(
  overrides: Partial<BillingRuntimeState> = {},
): BillingRuntimeState {
  return { ...BASE, ...overrides }
}

function noResources(
  overrides: Partial<BillingRuntimeState> = {},
): BillingRuntimeState {
  return {
    ...BASE,
    entitlementStatus: 'no_resources',
    remainingTotalTokens: 0,
    ...overrides,
  }
}

function trial(
  overrides: Partial<BillingRuntimeState> = {},
): BillingRuntimeState {
  return {
    ...BASE,
    currentSubscriptionStatus: 'trialing',
    trialEndsAt: '2030-01-01T00:00:00Z',
    ...overrides,
  }
}

function featureGrant(
  feature: string,
  privilege: string,
  value: boolean,
): BillingRuntimeFeature {
  return {
    code: feature,
    privileges: [{ code: privilege, value, valueType: 'boolean', config: {} }],
  }
}

function featureAllowed(
  feature: string,
  privilege: string,
  overrides: Partial<BillingRuntimeState> = {},
): BillingRuntimeState {
  return { ...BASE, features: [featureGrant(feature, privilege, true)], ...overrides }
}

function featureDenied(
  feature: string,
  privilege: string,
  overrides: Partial<BillingRuntimeState> = {},
): BillingRuntimeState {
  return { ...BASE, features: [featureGrant(feature, privilege, false)], ...overrides }
}

export const fixtures = {
  runtimeState: {
    active,
    no_resources: noResources,
    trial,
    feature_allowed: featureAllowed,
    feature_denied: featureDenied,
  },
}
